docs/basics.md

```diff
@@ -157,6 +157,153 @@ other things yourself, if needed.
+  // ...
+});
+```
+
+You can often achieve this with `process.browser` in most settings if
+you're using a single universal file to create a client on the server
+and on the client.
+
+Next up, the `ssrExchange` needs to be set up. It's a factory, since
+it has some methods for extracting and rehydrating data.
+
+```js
+import {
+  Client,
+  dedupExchange,
+  cacheExchange,
+  fetchExchange,
+  ssrExchange,
+} from 'urql';
+
+const ssrCache = ssrExchange();
+
+const client = new Client({
+  exchanges: [
+    dedupExchange,
+    cacheExchange,
+    ssrCache, // Added here after calling ssrExchange
+    fetchExchange,
+  ],
+  // ...
+  suspense: !process.browser,
+});
+```
+
+The exchange from `ssrExchange()` should be added after the default
+or your custom `cacheExchange` and before any asynchronous exchanges
+like the `fetchExchange`.
+
+### Prefetching on the server
+
+In your request handler on the server-side, you'll have to add some
+code handling suspense. Typically this is done using a "prepass" that
+walks your element tree and awaits suspended promises.
+
+So for suspense to now be executed on the server you may install
+`react-ssr-prepass` (which has a peer dependency on `react-is`)
+
+```sh
+yarn add react-ssr-prepass react-is
+# or
+npm install --save react-ssr-prepass react-is
+```
+
+Then you can add it to your server-side rendering code _before_
+calling `renderToString` or `renderToNodeStream`. This will fetch
+all suspended promises, including `urql`'s queries. And after
+you can use the `ssrExchange()`'s `extractData` method to
+get `urql`'s data:
+
+```js
+import ssrPrepass from 'react-ssr-prepass';
+
+const handler = (req, res) => {
+  // ...
+  // We assume you've already set up the urql Client and have
+  // the `ssrCache = ssrExchange()` variable from somewhere
+
+  await ssrPrepass(<App />);
+
+  // Extract the data from urql's SSR cache
+  const urqlData = ssrCache.extractData();
+
+  // Then you can run your rendering code for which the ssrCache
+  // should remain unchanged
+  const reactHtml = renderToString(<App />);
+
+  // Make sure to send urql's data down to the client somehow
+  const urqlHtml = `<script>window.URQL_DATA = ${JSON.stringify(urqlData)};</script>`;
+
+  // And send everything down to the client
+  // ...
+};
+```
+
+### Rehydrating on the client
+
+Assuming you now have server-side rendered and have sent down `urql`'s
+data from that render pass to the client, you should make sure that you
+rehydrate that data.
+
+You can either do so when creating `ssrExchange`, by passing it `initialState`
+as a parameter or calling `restoreData` on it:
+
+```js
+import { ssrExchange } from 'urql';
+
+const ssrCache = ssrExchange({
+  initialState: window.URQL_DATA,
+});
+
+// or:
+
+ssrCache.restoreData(window.URQL_DATA);
+
+// Assuming this follows the client setup and is added to the `exchanges` list
+// ...
+```
+
+Your setup may vary depending on whether you set up the same file to be
```

Comment:
Replace paragraph with:

Your setup may vary depending on whether your client initialization code is universal (single file that executes on client and server, such as Next.js `getInitialProps`), or two separate files for client and server.